cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/ReduceAggregator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api;
+
+import io.cdap.cdap.api.annotation.Beta;
+
+/**
+ * Groups all input objects into collections, and performs an aggregation on the entire group.
+ * Objects that have the same group key are placed into the same group for aggregation.
+ * It has better performance than {@link Aggregator} since the number of group values is reduced in each split before
```

Comment:
```
When possible, this interface should be used over the {@link Aggregator} interface because this performs better. 
An {@link Aggregator} will shuffle all data across the cluster before aggregating, whereas
this will aggregate both before and after the shuffle.
This reduces the amount of data that needs to be sent over the network, as well as reducing the amount
of memory required to perform the aggregation.
```